### ISSUE TITLE
CONTRIBUTING.md: Add section on becoming a maintainer

### ISSUE THREAD

**Issue by Mindavi** [NixOS/nixpkgs#171557]
I got some feedback after a review on a PR (https://github.com/NixOS/nixpkgs/pull/161404#issuecomment-1117081506) that it's not very clear that adding yourself as a maintainer in maintainers.nix should be a separate commit.

It might be nice to add a small section to CONTRIBUTING explaining the (simple) process, or maybe linking from there to the manual.

**Comment by fabaff** [comment]
The [Nixpkgs Manual](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers) has that section.

**Comment by VuiMuich** [comment]
Nevertheless the section in the manual is helpful, imo having a comment in CONTRIBUTING seems like a good idea.

Additionally I would suggest that both sources should mention that as [per comment](https://github.com/NixOS/nixpkgs/pull/251335) it might be preferrable to have the commits ordered like this:
```
maintainers: add <new-maintainer>
<package-name>: <commit message>
```
this also helps to easier squash required changes to the actual commit.

**Comment by fricklerhandwerk** [comment]
Fixed by #248013 and #250344
